insights/client/connection.py

```diff
@@ -299,167 +376,152 @@ def get_proxies(self):
-                print(exc)
-        if last_ex:
-            raise last_ex
+        test_url = url.scheme + "://" + url.netloc + url.path
+        logger.log(NETWORK, "Testing: %s", test_url)
+        if method is "POST":
+            test_req = self.post(test_url, data=test_flag)
+        elif method is "GET":
+            test_req = self.get(test_url)
+        if not test_req:
+            logger.error("Could not successfully connect to: %s", test_url)
+            return False
+        logger.info("Successfully connected to: %s", test_url)
+        return True
 
     def _test_urls(self, url, method):
         '''
-        Test a URL
+        Send a request to test a URL
+
+        Parameters
+            url     - URL to send a test request to
+            method  - HTTP method to use
+
+        Returns
+            True on success
+            False on failure
         '''
         if self.config.legacy_upload:
             return self._legacy_test_urls(url, method)
-        try:
-            logger.log(NETWORK, 'Testing %s', url)
-            if method is 'POST':
-                test_tar = TemporaryFile(mode='rb', suffix='.tar.gz')
-                test_files = {
-                    'file': ('test.tar.gz', test_tar, 'application/vnd.redhat.advisor.collection+tgz'),
-                    'metadata': '{\"test\": \"test\"}'
-                }
-                test_req = self.session.post(url, timeout=self.config.http_timeout, files=test_files)
-            elif method is "GET":
-                    test_req = self.session.get(url, timeout=self.config.http_timeout)
-            logger.log(NETWORK, "HTTP Status Code: %d", test_req.status_code)
-            logger.log(NETWORK, "HTTP Status Text: %s", test_req.reason)
-            logger.log(NETWORK, "HTTP Response Text: %s", test_req.text)
-            if test_req.status_code in (200, 201, 202):
-                logger.info(
-                    "Successfully connected to: %s", url)
-                return True
-            else:
-                logger.info("Connection failed")
-                return False
-        except requests.ConnectionError as exc:
-            last_ex = exc
-            logger.error(
-                "Could not successfully connect to: %s", url)
-            print(exc)
-        if last_ex:
-            raise last_ex
-
-    def test_connection(self, rc=0):
+        logger.log(NETWORK, 'Testing %s', url)
+        if method is 'POST':
+            test_tar = TemporaryFile(mode='rb', suffix='.tar.gz')
+            test_files = {
+                'file': ('test.tar.gz', test_tar, 'application/vnd.redhat.advisor.collection+tgz'),
+                'metadata': '{\"test\": \"test\"}'
+            }
+            test_req = self.post(url, files=test_files)
+        elif method is "GET":
+            test_req = self.get(url)
+        if not test_req:
+            logger.error("Could not successfully connect to: %s", url)
```

Comment:
Just to be clear. This exact logging call used to be used on a connection error, which is not a case of a non-success status code. Now, this is logged instead of `logger.info("Connection failed")`. It looks like a good change, just saying.